### PR TITLE
A moderate amount of fixes

### PR DESCRIPTION
**Pull request by kelemeno**
legacy eth address
l2TokenAddress fix
eth deposit gas estimation and erc20.name fix
New rpc in provider for l2BaseToken
setup and test changes

# What :computer: 
* First thing updated with this PR
* Second thing updated with this PR
* Third thing updated with this PR

# Why :hand:
* Reason why first thing was added to PR
* Reason why second thing was added to PR
* Reason why third thing was added to PR

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

# Notes :memo:
* Any notes/thoughts that the reviewers should know prior to reviewing the code?

### CODE REVIEW

**Review by Raid5594 — APPROVED** [review]

File: tests/integration/wallet.test.ts [hunk 1]
```diff
@@ -471,13 +480,13 @@ describe("Wallet", async () => {
...
     describe("#deposit()", () => {
-        if (isETHBasedChain) {
+        if (isETHBased) {
             it("should deposit ETH to L2 network", async () => {
                 const amount = 7_000_000_000;
                 const l2BalanceBeforeDeposit = await wallet.getBalance();
                 const l1BalanceBeforeDeposit = await wallet.getBalanceL1();
                 const tx = await wallet.deposit({
-                    token: utils.ETH_ADDRESS,
+                    token: utils.LEGACY_ETH_ADDRESS,
```
Comment (Raid5594) [inline tests/integration/wallet.test.ts:489]: How did the gas estimation changed here?

File: tests/integration/wallet.test.ts [hunk 2]
```diff
@@ -595,10 +604,10 @@ describe("Wallet", async () => {
     });
 
     describe("#claimFailedDeposit()", () => {
-        if (isETHBasedChain) {
+        if (isETHBased) {
             // it("should claim failed deposit", async () => {
```
Comment (Raid5594) [inline tests/integration/wallet.test.ts:608]: This should be uncommented as deposit now works right?

File: src/utils.ts [hunk 3]
```diff
@@ -550,7 +550,7 @@ export async function estimateDefaultBridgeDepositL2Gas(
     // due to storage slot aggregation, the gas estimation will depend on the address
     // and so estimation for the zero address may be smaller than for the sender.
     from ??= ethers.Wallet.createRandom().address;
-    if (isBaseTokenDeposit || token === ETH_ADDRESS) {
```
Comment (kelemeno) [inline src/utils.ts:553]: Changed the estimation here mostly,` token === Eth_address` was there incorrectly. Also the :
```
const name = l1TokenAddress == ETH_ADDRESS_IN_CONTRACTS ? "Ether" :  await token.name();
    const symbol =  l1TokenAddress == ETH_ADDRESS_IN_CONTRACTS ? "ETH" : await token.symbol();
    const decimals =  l1TokenAddress == ETH_ADDRESS_IN_CONTRACTS ? 18 :  await token.decimals();
```

Comment (kelemeno) [reply src/utils.ts:553]: @Raid5594 , please close if ok

File: src/utils.ts [hunk 4]
```diff
@@ -28,13 +28,13 @@ export const L1_BRIDGE_ABI = new utils.Interface(require("../abi/IL1Bridge.json"
 export const L2_BRIDGE_ABI = new utils.Interface(require("../abi/IL2Bridge.json"));
 export const NONCE_HOLDER_ABI = new utils.Interface(require("../abi/INonceHolder.json"));
 
-export const ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
```
Comment (danijelTxFusion) [inline src/utils.ts:31]: This constant is part of public API, renaming it is a breakable change. There are 2 solutions:
- keep it as it is,
- mark it as deprecated, introduce new constant `LEGACY_ETH_ADDRESS` and using it instead.

File: src/utils.ts [hunk 5]
```diff
@@ -28,13 +28,13 @@ export const L1_BRIDGE_ABI = new utils.Interface(require("../abi/IL1Bridge.json"
...
+export const LEGACY_ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
 // in the contracts the zero address can not be used, use one instead
 export const ETH_ADDRESS_IN_CONTRACTS = "0x0000000000000000000000000000000000000001";
 export const BOOTLOADER_FORMAL_ADDRESS = "0x0000000000000000000000000000000000008001";
 export const CONTRACT_DEPLOYER_ADDRESS = "0x0000000000000000000000000000000000008006";
 export const L1_MESSENGER_ADDRESS = "0x0000000000000000000000000000000000008008";
-export const L2_ETH_TOKEN_ADDRESS = "0x000000000000000000000000000000000000800a";
```
Comment (danijelTxFusion) [inline src/utils.ts:37]: This is also part of public API. In this case I am to mark it as deprecated and use new constant `L2_BASE_TOKEN_ADDRESS` to be more intuitive.

File: src/utils.ts (hunk 4 above)
Comment (kelemeno) [reply src/utils.ts:31]: Ok, we will have:
ETH_ADDRESS = 0x00
LEGACY_ETH_ADDRESS = 0X00
ETH_ADDRESS_IN_CONTRACTS = 0X01.

File: typechain/Il1Bridge.d.ts
Comment (danijelTxFusion) [inline typechain/Il1Bridge.d.ts:1]: This needs to stay and mark it as deprecated.

**Pull request merged**